packages/std/src/math/conversion.rs

```diff
@@ -0,0 +1,100 @@
+/// Grows a big endian signed integer to a bigger size.
+pub const fn grow_be_int<const INPUT_SIZE: usize, const OUTPUT_SIZE: usize>(
+    input: [u8; INPUT_SIZE],
+) -> [u8; OUTPUT_SIZE] {
+    debug_assert!(INPUT_SIZE <= OUTPUT_SIZE);
+    // check if sign bit is set
+    let mut output = if input[0] & 0b10000000 != 0 {
+        // negative number is filled up with 1s
+        [0b11111111u8; OUTPUT_SIZE]
+    } else {
+        [0u8; OUTPUT_SIZE]
+    };
+    let mut i = 0;
+
+    // copy input to the end of output
+    // copy_from_slice is not const, so we have to do this manually
+    while i < INPUT_SIZE {
+        output[OUTPUT_SIZE - INPUT_SIZE + i] = input[i];
+        i += 1;
+    }
+    output
+}
+
+/// Shrinks a big endian signed integer to a smaller size.
+pub fn shrink_be_int<const INPUT_SIZE: usize, const OUTPUT_SIZE: usize>(
+    input: [u8; INPUT_SIZE],
+) -> Option<[u8; OUTPUT_SIZE]> {
+    debug_assert!(INPUT_SIZE >= OUTPUT_SIZE);
+
+    // A positive number should start with only 0s and a negative one with only 1s until
+    // the size we want to look at.
+    // If this is not the case, then the value is too large / small for the target type
+    let ignored_byte = if input[0] & 0b10000000 != 0 {
+        0b11111111u8
+    } else {
+        0u8
+    };
+    // Rust doesn't allow us to create an array of size `OUTPUT_SIZE - INPUT_SIZE`,
+    // so we work around this by taking a slice of a bigger array
+    let ignore_bytes = [ignored_byte; INPUT_SIZE];
+    if input[0..(INPUT_SIZE - OUTPUT_SIZE)] != ignore_bytes[0..(INPUT_SIZE - OUTPUT_SIZE)] {
+        return None;
+    }
```

Comment:
Instead of constructing the `ignored_byte` array, can't you just say

```
if input[0] & 0b10000000 != 0 {
      // loop through input[0..(INPUT_SIZE - OUTPUT_SIZE)] and return None if byte is not 0b11111111u8
} else {
      // loop through input[0..(INPUT_SIZE - OUTPUT_SIZE)] and return None if byte is not 0u8
}
```

I assume this would make the code clearer. But both ways are fine.

Reply:
Good point, it's a lot more readable that way